libs/common/src/dtfs-central-api-response-types/utilisation-reports.ts

```diff
@@ -0,0 +1,19 @@
+import { UtilisationReportReconciliationStatus, ReportPeriod, IsoDateTimeStamp } from '..';
+
+export type AzureFileInfo = {
+  folder: string;
+  filename: string;
+  fullPath: string;
+  url: string;
+  mimetype: string;
+};
+
+export type UtilisationReportResponseBody = {
+  id: number;
+  bankId: string;
+  reportPeriod: ReportPeriod;
+  dateUploaded: IsoDateTimeStamp;
+  azureFileInfo: AzureFileInfo | null;
+  status: UtilisationReportReconciliationStatus;
+  uploadedByUserId: string | null;
+};
```

Comment:
In order to keep a link between this type and `UtilisationReportEntity`, I think we should use something like this

```suggestion
type UtilisationReportResponseBody2 = Pick<UtilisationReportEntity, 'id' | 'bankId' | 'reportPeriod' | 'status' | 'uploadedByUserId'> & {
  azureFileInfo: AzureFileInfo | null;
  dateUploaded: IsoDateTimeStamp;
};
```

Reply:
I have never really made much use of Omit and Pick before... Seems like a good idea, though when I over on it in vscode then it is rather unclear what the types of the individual fields are. I don't know if that is a good enough reason to not do it.
I guess another reason is that in some respects it is a coincidence they match rather than a cause, in that we might want our db to store stuff differently to how we want to return it from the api and so in tying them together we are perhaps making a stronger link than there really should be?

Reply:
We have a `Prettify` utility type to get around the clarity issue when hovering on the type. That being said, I think the simplicity of the type in this case means that hovering on the type using `Pick` I've created above does show the "prettified" version anyway.

I don't think it is a coincidence that the types match up. For example, `status` should always like up with the type of `UtilisationReportEntity['status']`. I think it makes it clearer if we use a `Pick` and an intersection like I have done to make it obvious which parameters have been messed with or changed and which parameters have been left as they are in the database

Reply:
FWIW I find this conversation quite interesting.
On one hand if we use omit then it means any new fields that will naturally be in both types get added automatically.
On the other hand if there is some random database type we add that is generated and we don't want it on this type then we have to add it to an omit to not break stuff.
In both those cases it means that there's pros and cons so it's hard to know the best choice - for some things like an API response where we mostly just return database like types does that mean we should go for omit?
In other APIs such as graphql the API response style types are normally pretty separate from say your database types and they don't overlap so much